src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -106,29 +139,21 @@ const HeatmapPlot = (props) => {
     setIsHeatmapGenesLoading(false);
   }, [selectedGenes, loadingGenes, markerGenesLoading]);
 
-  useEffect(() => {
-    if (cellSetsLoading || cellSetsHierarchy.length === 0) {
-      return;
-    }
-
-    const legends = legendIsVisible ? spec.legends : [];
-    setVegaSpec({ ...spec, legends });
-  }, [legendIsVisible]);
-
   useEffect(() => {
```

Comment:
somewhere here you have a useEffect that gets triggered if either of the following changes:

`[selectedGenes, loadingGenes, markerGenesLoading]`

What is the technical difference between `loadingGenes` and `markerGenesLoading`? Aren't `markerGenesLoading` just a type of `loadingGenes`?

Reply:
They are 2 different worker requests: `markerGenesLoading` is for loading the top N marker genes and `loadingGenes` for loading genes in particular